pkg/util/procfs/procfs_linux.go

```diff
@@ -105,27 +105,29 @@ func PidOf(name string) ([]int, error) {
 
 func getPids(re *regexp.Regexp) []int {
 	pids := []int{}
-	filepath.Walk("/proc", func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			// We should continue processing other directories/files
-			return nil
-		}
-		base := filepath.Base(path)
-		// Traverse only the directories we are interested in
-		if info.IsDir() && path != "/proc" {
-			// If the directory is not a number (i.e. not a PID), skip it
-			if _, err := strconv.Atoi(base); err != nil {
-				return filepath.SkipDir
-			}
+
+	ls, err := ioutil.ReadDir("/proc")
+	if err != nil {
+		return nil
+	}
+
+	for _, entry := range ls {
+		if !entry.IsDir() {
+			continue
 		}
-		if base != "cmdline" {
-			return nil
+
+		// If the directory is not a number (i.e. not a PID), skip it
+		pid, err := strconv.Atoi(entry.Name())
+		if err != nil {
+			continue
 		}
-		cmdline, err := ioutil.ReadFile(path)
+
+		cmdline, err := ioutil.ReadFile(filepath.Join("/proc", entry.Name(), "cmdline"))
 		if err != nil {
-			glog.V(4).Infof("Error reading file %s: %+v", path, err)
-			return nil
+			glog.V(4).Infof("Error reading file %s: %+v", filepath.Join("/proc", entry.Name(), "cmdline"), err)
+			continue
 		}
+
 		// The bytes we read have '\0' as a separator for the command line
 		parts := bytes.SplitN(cmdline, []byte{0}, 2)
 		if len(parts) == 0 {
```

Comment:
Should be `continue` rather than `return`.

Reply:
Good catch.